gensim/models/word2vec.py

```diff
@@ -1191,383 +1191,78 @@ def intersect_word2vec_format(self, fname, lockf=0.0, binary=False, encoding='ut
+        if item == "syn0":
+            return self.kv.syn0
+        if item == "syn0norm":
+            return self.kv.syn0norm
+        if item == "vocab":
+            return self.kv.vocab
+        if item == "index2word":
+            return self.kv.index2word
+        raise AttributeError("'{0}' object has no attribute '{1}'".format(type(self).__name__, item))
 
     def __contains__(self, word):
-        return word in self.vocab
+        return self.kv.__contains__(word)
 
     def similarity(self, w1, w2):
-        """
-        Compute cosine similarity between two words.
-
-        Example::
-
-          >>> trained_model.similarity('woman', 'man')
-          0.73723527
-
-          >>> trained_model.similarity('woman', 'woman')
-          1.0
-
-        """
-        return dot(matutils.unitvec(self[w1]), matutils.unitvec(self[w2]))
+        return self.kv.similarity(w1, w2)
 
     def n_similarity(self, ws1, ws2):
-        """
-        Compute cosine similarity between two sets of words.
-
-        Example::
-
-          >>> trained_model.n_similarity(['sushi', 'shop'], ['japanese', 'restaurant'])
-          0.61540466561049689
-
-          >>> trained_model.n_similarity(['restaurant', 'japanese'], ['japanese', 'restaurant'])
-          1.0000000000000004
-
-          >>> trained_model.n_similarity(['sushi'], ['restaurant']) == trained_model.similarity('sushi', 'restaurant')
-          True
-
-        """
-        v1 = [self[word] for word in ws1]
-        v2 = [self[word] for word in ws2]
-        return dot(matutils.unitvec(array(v1).mean(axis=0)), matutils.unitvec(array(v2).mean(axis=0)))
+        return self.kv.n_similarity(ws1, ws2)
 
     def init_sims(self, replace=False):
         """
-        Precompute L2-normalized vectors.
-
-        If `replace` is set, forget the original vectors and only keep the normalized
-        ones = saves lots of memory!
-
-        Note that you **cannot continue training** after doing a replace. The model becomes
-        effectively read-only = you can call `most_similar`, `similarity` etc., but not `train`.
-
+        init_sims() resides in KeyedVectors because it deals with syn0 mainly, but because syn1 is not an attribute
+        of KeyedVectors, it has to be deleted in this class, and the normalizing of syn0 happens inside of KeyedVectors
         """
-        if getattr(self, 'syn0norm', None) is None or replace:
-            logger.info("precomputing L2-norms of word weight vectors")
-            if replace:
-                for i in xrange(self.syn0.shape[0]):
-                    self.syn0[i, :] /= sqrt((self.syn0[i, :] ** 2).sum(-1))
-                self.syn0norm = self.syn0
-                if hasattr(self, 'syn1'):
-                    del self.syn1
-            else:
-                self.syn0norm = (self.syn0 / sqrt((self.syn0 ** 2).sum(-1))[..., newaxis]).astype(REAL)
+        if hasattr(self, 'syn1'):
+            del self.syn1
```

Comment:
`replace` still needs to be checked. Otherwise, a call to `init_sims` even with `replace=False` will cause the model to become incapable of continuing training.